libcontainerd/supervisor/remote_daemon.go

```diff
@@ -245,18 +252,28 @@ func (r *remote) monitorDaemon(ctx context.Context) {
 		r.platformCleanup()
 
 		close(r.daemonStopCh)
+
+		if timer != nil {
+			timer.Stop()
+		}
 	}()
 
 	for {
-		if delay != nil {
+		if delay > 0 {
+			if timer == nil {
+				timer = time.NewTimer(delay)
+			}
+
+			timer.Reset(delay)
```

Comment:
Quoting official docs, I believe it might be applicable here:

> Resetting a timer must take care not to race with the send into t.C that happens when the current timer expires. If a program has already received a value from t.C, the timer is known to have expired, and t.Reset can be used directly. If a program has not yet received a value from t.C, however, the timer must be stopped and—if Stop reports that the timer expired before being stopped—the channel explicitly drained:
```go
if !t.Stop() {
	<-t.C
}
t.Reset(d)
```